packages/twenty-front/src/modules/views/components/QueryParamsViewIdEffect.tsx

```diff
@@ -1,120 +1,34 @@
 import { contextStoreCurrentViewIdComponentState } from '@/context-store/states/contextStoreCurrentViewIdComponentState';
-import { useLastVisitedObjectMetadataItem } from '@/navigation/hooks/useLastVisitedObjectMetadataItem';
-import { useLastVisitedView } from '@/navigation/hooks/useLastVisitedView';
-import { useFilteredObjectMetadataItems } from '@/object-metadata/hooks/useFilteredObjectMetadataItems';
 import { useRecoilComponentStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentStateV2';
-import { useSetRecoilComponentStateV2 } from '@/ui/utilities/state/component-state/hooks/useSetRecoilComponentStateV2';
-import { useViewFromQueryParams } from '@/views/hooks/internal/useViewFromQueryParams';
-import { useGetCurrentView } from '@/views/hooks/useGetCurrentView';
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
 import { currentViewIdComponentState } from '@/views/states/currentViewIdComponentState';
-import { isUndefined } from '@sniptt/guards';
 import { useEffect } from 'react';
 import { isDefined } from 'twenty-shared';
-import { isDeeplyEqual } from '~/utils/isDeeplyEqual';
-
-export const QueryParamsViewIdEffect = () => {
-  const { getFiltersFromQueryParams, viewIdQueryParam } =
-    useViewFromQueryParams();
-
-  // TODO: fix this implicit hack
-  const { instanceId: objectNamePlural } = useGetCurrentView();
 
+// TODO: This whole code should be removed. currentViewId should be used directly to set the mainContextStore
+// and viewbar / view tooling should be updated to use that state contextStore state directly.
+export const QueryParamsViewIdEffect = ({
+  objectNamePlural,
+}: {
+  objectNamePlural: string;
+}) => {
   const [currentViewId, setCurrentViewId] = useRecoilComponentStateV2(
     currentViewIdComponentState,
   );
 
-  const { viewsOnCurrentObject } = useGetCurrentView();
```

Comment:
this whole logic was duplicated with ContextStoreViewIdEffect. I'm just syncing currentViewId with contextStoreCurrentViewId for now but see TODO comment